### PR TITLE
nbt: ListBinaryTag.add(Iterable)

### PR DESCRIPTION
**Pull request by kashike**

### CODE REVIEW

File: nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTagImpl.java [hunk 1]
```diff
@@ -64,34 +64,42 @@ public int size() {
...
+    noAddEnd(tag);
+    if(this.elementType != BinaryTagTypes.END) {
+      mustBeSameType(tag, this.elementType);
+    }
+    return this.edit(tags -> tags.add(tag), tag.type());
+  }
+
+  @Override
+  public @NonNull ListBinaryTag add(final Iterable<? extends BinaryTag> tagsToAdd) {
+    final BinaryTagType<?> type = ListBinaryTagImpl.mustBeSameType(tagsToAdd);
```
Comment (zml2008) [inline nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTagImpl.java:97]: add a check here that `tagsToAdd` is non-empty maybe, and if it is empty return `this` to save the copy?

File: nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTag.java [hunk 2]
```diff
@@ -46,6 +46,16 @@
     return ListBinaryTagImpl.EMPTY;
   }
 
+  /**
+   * Gets an empty list tag.
```
Comment (zml2008) [inline nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTag.java:50]: you forgot to update this :p

File: nbt/src/main/java/net/kyori/adventure/nbt/ListTagSetter.java [hunk 3]
```diff
@@ -41,4 +41,13 @@
    * @since 4.0.0
    */
   @NonNull R add(final T tag);
+
+  /**
+   * Adds tags.
```
Comment (zml2008) [inline nbt/src/main/java/net/kyori/adventure/nbt/ListTagSetter.java:46]:
```suggestion
* Adds multiple tags
```

**Review by zml2008 — CHANGES_REQUESTED** [review]

File: nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTagImpl.java (hunk 1 above)
Comment (kashike) [reply nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTagImpl.java:97]: `Iterable` has no `isEmpty` or `size` method?

Comment (zml2008) [reply nbt/src/main/java/net/kyori/adventure/nbt/ListBinaryTagImpl.java:97]: you could either create an Iterator and check hasNext before going into `edit()`, or only do the optimization for inputs that are also a `Collection` i guess

**Pull request merged**